frontend/src/common/queries/collections.ts

```diff
@@ -248,20 +263,46 @@ async function deleteCollection(collectionID: Collection["id"]) {
   }
 }
 
-export function useDeleteCollection(id = "") {
-  if (!id) {
-    throw new Error("No collection id given");
-  }
-
+export function useDeleteCollection(
+  id = "",
+  visibility = VISIBILITY_TYPE.PRIVATE
+) {
   const queryCache = useQueryCache();
 
   return useMutation(deleteCollection, {
+    onError: (
+      _,
+      __,
+      context: { previousCollections: CollectionResponsesMap }
+    ) => {
+      queryCache.setQueryData([USE_COLLECTIONS], context.previousCollections);
+    },
+    onMutate: async () => {
+      await queryCache.cancelQueries([USE_COLLECTIONS]);
+
+      const previousCollections = queryCache.getQueryData([
+        USE_COLLECTIONS,
+      ]) as CollectionResponsesMap;
+
+      const newCollections = new Map(previousCollections);
+      // If we're deleting a public collection, nuke it from the cache
+      if (visibility === VISIBILITY_TYPE.PUBLIC) {
```

Comment:
Actually there's a second case where we nuke it, when there is only a private collection. Fixing